esrally/client/__init__.py

```diff
@@ -0,0 +1,19 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from .client import EsClientFactory, wait_for_rest_layer
```

Comment:
I went back and forth a bit on whether to expose these under the `esrally.client` namespace (as is done here) or just have callers import from `esrally.client.client` instead. I don't have a strong leaning here. Curious to hear what you think @pquentin.

Reply:
Reexporting is fine, callers should not have to worry about the way `esrally.client` is organized internally.

However `client.client` is a bit confusing, should we call it `client.factory`? (Since this is internal to `esrally.client` it's easuyto change :)